### PR TITLE
wio publish

### PR DESCRIPTION
**Pull request by Mogball**

### CODE REVIEW

File: cmd/wio/commands/pac/user/login.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+func (cmd Login) getArgs() (*loginArgs, error) {
+    ctx := cmd.Context
+    dir, err := commands.GetDirectory(cmd)
+    if err != nil {
+        return nil, err
+    }
+    args := ctx.Args()
+    if len(args) < 3 {
+        return nil, errors.New("wio login [username] [password] [email]")
```
Comment (dhillondeep) [inline cmd/wio/commands/pac/user/login.go]: we should make password masked like *****. 

https://github.com/howeyc/gopass
https://stackoverflow.com/questions/2137357/getpasswd-functionality-in-go

File: cmd/wio/commands/pac/user/logout.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "wio/cmd/wio/log"
+    "wio/cmd/wio/utils/io"
+)
+
+func (cmd Logout) Execute() error {
+    dir, err := commands.GetDirectory(cmd)
+    if err != nil {
+        return nil
+    }
+    path := io.Path(dir, io.Folder, "token.json")
```
Comment (dhillondeep) [inline cmd/wio/commands/pac/user/logout.go:16]: this means you will have a different token based on your wio project. Can you log out if you delete that directory or change wio projects?

Comment (dhillondeep) [reply cmd/wio/commands/pac/user/logout.go:16]: Oops deleting the token means logging out

Comment (dhillondeep) [reply cmd/wio/commands/pac/user/logout.go:16]: isn't storing token this way a security risk? lol

File: cmd/wio/toolchain/npm/publish/types.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+)
+
+type Attachment struct {
+    Type   string `json:"content-type"`
+    Data   string `json:"data"`
+    Length uint64 `json:"length"`
+}
+
+type Data struct {
+    Id          string `json:"_id"`
```
Comment (dhillondeep) [inline cmd/wio/toolchain/npm/publish/types.go:15]: should this be _id or id?

File: cmd/wio/commands/pac/user/logout.go (hunk 2 above)
Comment (Mogball) [reply cmd/wio/commands/pac/user/logout.go:16]: how do you think `npm`, `git`, `pip` and literally every other package manager does it?

the token has an expiry

File: cmd/wio/toolchain/npm/publish/types.go (hunk 3 above)
Comment (Mogball) [reply cmd/wio/toolchain/npm/publish/types.go:15]: it's pretty hard to accidentally insert an underscore

File: cmd/wio/commands/pac/user/login.go (hunk 1 above)
Comment (Mogball) [reply cmd/wio/commands/pac/user/login.go]: was too lazy to add command-line interface you can do it :)

File: cmd/wio/toolchain/npm/publish/types.go (hunk 3 above)
Comment (dhillondeep) [reply cmd/wio/toolchain/npm/publish/types.go:15]: true

File: cmd/wio/commands/pac/user/logout.go (hunk 2 above)
Comment (dhillondeep) [reply cmd/wio/commands/pac/user/logout.go:16]: ik but the token is in `.wio` directory so it is only valid as long as I am in that specific wio project. Right?

File: cmd/wio/commands/pac/user/login.go (hunk 1 above)
Comment (dhillondeep) [reply cmd/wio/commands/pac/user/login.go]: lol fine

**Pull request merged**